### PR TITLE
Added extra iam permissions to work with existing log group

### PR DESCRIPTION
**Pull request by tylernap**
We currently have a log group that already exists with a function that already exists. This adds permissions to those already existing log groups. This will not impact new deployments.

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -22,6 +22,17 @@ provider:
...
+          Action:
+           - logs:CreateLogStream
+           - logs:CreateLogGroup
+          Resource:
+           - "arn:aws:iam::${env:ACCOUNT}:log-group:/aws/lambda/cloudflare-lambda-Function-*:*"
+        - Effect: Allow
+          Action:
+           - logs:PutLogEvents
+          Resource:
+           - "arn:aws:iam::${env:ACCOUNT}:log-group:/aws/lambda/cloudflare-lambda-Function-*:*:*"
```
Comment (gorskilauren) [inline serverless.yml:35]: What does the extra `*:*:*` here (vs. `*:*` on line 30) in this resource arn represent? Moreover if we know the existing "old" loggroup this lambda is trying to log to could we specify that arn and get rid of the wildcard(s) altogether in the spirit of least privilege access?

Comment (tylernap) [reply serverless.yml:35]: This line differs from line 30 because log events are generated within log groups. Line 30 uses `/aws/lambda/cloudflare-lambda-Function-*:*` to have the ability to make a log group of any name. Line 35 uses `/aws/lambda/cloudflare-lambda-Function-*:*:*` because it needs to be allowed to make log events in the log groups that get created.

We could certainly just hardcode the lambda name, but since this is something we have out on Github and is meant to be used by people outside of our company, I figured it would be best to simply wildcard it.

Comment (gorskilauren) [reply serverless.yml:35]: Hmm.. I wonder if we could reference it [sort of like this](https://github.com/Dwolla/cloudflare-lambda/blob/fix-lambda-iam/serverless.yml#L74-L78) so it wouldn't need to be hardcoded to our specific function name:
```
functions:
  Cloudflare:
    handler: com.dwolla.lambda.cloudflare.Handler
    name: cloudflare-lambda
Resource:
    - self:custom.FUNCTION_LOG_GROUP_ARN
custom:
  FUNCTION_LOG_GROUP_ARN:
    Fn::Join:
      - ''arn:aws:iam::${env:ACCOUNT}:log-group:/aws/lambda/cloudflare-lambda-Function-"
      - Fn::GetAtt: [CloudflareLambdaFunction, Name]
```

(there might be an easier way to interpolate the function name onto the log group, this is just from a [stackoverflow example](https://stackoverflow.com/questions/64350072/serverless-variable-interpolation-envvar-fallback-using-fngetatt) I found. Also log group might even be an attribute on CloudflareLambdaFunction, though I didn't see it in the [serverless docs](https://www.serverless.com/framework/docs/providers/aws/guide/functions))

Comment (tylernap) [reply serverless.yml:35]: The problem I see with that is that you create a circular dependency. The role needs to have the function name and the function needs to have the role. Cloudformation would just error out.

I understand wanting least privilege which is why I added as much of the existing function name as possible without revealing too much. Unless there is another lambda with a log group named `cloudflare-lambda-Function-somethingelse`, I don't see this stepping on any other toes. I want to avoid overengineering this one-off caveat that doesn't impact people outside of our org.

**Review by gorskilauren — APPROVED** [review]

File: serverless.yml (hunk 1 above)
Comment (tylernap) [reply serverless.yml:35]: So in our case, we had this lambda deployed previously without using serverless. After it was deployed, its exports were referenced by other stacks which means that now if we need to redeploy the lambda, we would need to remove all of those references before it gets recreated. We would need to redeploy the lambda if we changed the logical ID in the stack, which is what is happening since serverless uses a different logical ID scheme. 

To overcome this, a serverless deploy plugin was created to [alter the logical ID](https://github.com/Dwolla/cloudflare-lambda/blob/main/project/ServerlessDeployPlugin.scala#L39) after building. This leaves the existing lambda intact with the same ARN and same configurations. Since the lambda is referencing the same log groups, it is pointing to the old `cloudflare-lambda-Function-X` log group which is different than what serverless framework expects.

Basically, serverless doesn't understand the past of the lambda if it is taking over an existing lambda and continues deploying stuff as if it is completely under its control. It's a product of us switching to using serverless for a lambda stack that had cloudformation exports.
